great3sims/psf.py

```diff
@@ -1393,64 +1405,64 @@ def generateFieldParameters(self, rng, field_index):
         return psf_dict
 
     def generateEpochParameters(self, rng, subfield_index, epoch_index, field_parameters):
-        schema = [("psf_number", int),]
+        #   These schema members pass info about the Psf used and how to find it.
+        schema = [("psf_number", int), ("psf_index", int), ("psf_group", int),]
         psf_dict = dict(schema=schema)
         return psf_dict
 
     def generateCatalog(self, rng, catalog, parameters, offsets, normalized):
-
-        # The catalog will contain an entry for each record 'psf_number'
-        # corresponding to the random psf assigned to this galaxy.  In
-        # this routine, the psfs are generated if they come from a single
-        # multiple hdu catalog.  But if the draw_psf_src file is not found,
-        # assume that the psfs will come from numbered files in the directory
-        # where the draw_psf_src file is supposed to be. In either case, if the
-        # draw_psf_src is named foo.fits, the psfs are named foo_n.fits.
-       
-        index = self.draw_psf_src.find('.fits')
-        if index < 0: format = self.draw_psf_src + '_%d'
-        else:
-            format = self.draw_psf_src[:index] + '_%d' + self.draw_psf_src[index:]
-
-        # if the file self.draw_psf_src exists, assume it is a multi-hdu
-        # file containing a collection of psfs, and open if for access
-        hdus = None
-        if os.path.exists(self.draw_psf_src):
-            hdus = pyfits.open(self.draw_psf_src)
-            psf_count = len(hdus)
-
-        # if not, just enumerate the individual numbered fits files in
-        # the same directory
-        else:
-            psf_count = 0
-            while os.path.exists(format%(psf_count+1,)):
-                psf_count += 1
-            print "psf_count = ", psf_count
-        # now randomly assign availabel psfs to the individual records,
-        # expanding those which are not already there from the hdus
+        #   The psfInfos structure is a list of all the available Psfs,
+        #   Randomly assign available psfs to the individual records, and mark each
+        #   one which is choosen for expansion.  Note that the "psf_group" number is
+        #   another way to refer to the file named psfInfo[0], but using a number.
         for record in catalog:
             # select a random hdu
-            psf_number = int(math.floor(rng() * psf_count)) + 1 
-            record["psf_number"] = psf_number
-            filename = format%psf_number
-            if not os.path.exists(filename):
-                psf = pyfits.PrimaryHDU(hdus[psf_number-1].data)
+            psfInfo = self.psfInfos[int(math.floor(rng() * len(self.psfInfos)))]
+            record["psf_group"] = psfInfo[1]
+            record["psf_index"] = psfInfo[2]
+            record["psf_number"] = psfInfo[3]
+            psfInfo[4] = True
```

Comment:
So I'm guessing ``psfInfo[4]`` is a boolean indicating whether this particular PSF was randomly selected. It'd be nice to document that somewhere.